simtools/data_model/validate_data.py

```diff
@@ -62,39 +49,48 @@ def transform(self):
 
         """
 
-        self._check_data_for_duplicates()
-        self._sort_data()
+        self.validate_data_file()
+        if self._reference_data_columns is not None:
+            self._validate_data_columns()
+            self._check_data_for_duplicates()
+            self._sort_data()
 
         return self.data_table
 
     def validate_data_file(self):
         """
         Open data file and read data from file
+        (doing this successfully is understood as
+        file validation).
+
         """
 
         self._logger.info(f"Reading data from {self._data_file_name}")
         self.data_table = Table.read(self._data_file_name, guess=True, delimiter=r"\s")
```

Comment:
If you lost context, this is not a good sign. I know that the code is too complex!

I had QTables first, but that didn't work for certain cases (I don't have the log files anymore). I would suggest to leave it as Table (as every QTable is a Table).

Reply:
No problem!
And I lost context not necessarily because the code is complex but because sometimes when GitHub shows me only the changed stuff I am lazy to go back and read everything around it to make sure I understand the context before commenting. This is a symptom of long PRs, but not a real issue other than creating noise.